src/Adapters/Auth/ldap.js

```diff
@@ -0,0 +1,93 @@
+const ldapjs = require('ldapjs');
+const Parse = require('parse/node').Parse;
+
+function validateAuthData(authData, options) {
+  if (!optionsAreValid(options)) {
+    throw new Parse.Error(
+      Parse.Error.INTERNAL_SERVER_ERROR,
+      'LDAP auth configuration missing'
+    );
+  }
+
+  const client = ldapjs.createClient({ url: options.url });
+  const userCn =
+    typeof options.dn === 'string'
+      ? options.dn.replace('{{id}}', authData.id)
+      : `uid=${authData.id},${options.suffix}`;
+
+  return new Promise((resolve, reject) => {
+    client.bind(userCn, authData.password, err => {
+      if (err) {
+        client.destroy(err);
+        return reject(
+          new Parse.Error(
+            Parse.Error.OBJECT_NOT_FOUND,
+            'LDAP: Wrong username or password'
+          )
+        );
+      }
+
+      if (
+        typeof options.groupCn === 'string' &&
+        typeof options.groupFilter === 'string'
+      ) {
+        const filter = options.groupFilter.replace(/{{id}}/gi, authData.id);
+        const opts = {
+          scope: 'sub',
+          filter: filter,
+        };
+        let found = false;
+        client.search(options.suffix, opts, (searchError, res) => {
+          if (searchError) {
+            throw new Parse.Error(
```

Comment:
Can you add a test for this error?

Reply:
Added a test and found out that I forgot to destroy the client when this error happens...

Reply:
Can you just double check if you are destroying / unbinding the client in the correct places? If everything is good we can get this merged.